prober/dns.go

```diff
@@ -31,15 +31,15 @@ type ForEachAddrOpts struct {
 }
 
 // ForEachAddr returns a Probe that resolves a given hostname into all
-// available IP addresses, and then calls a function to create a new Probe
-// every time a new IP is discovered. The Probe returned will be closed if an
+// available IP addresses, and then calls a function to create new Probes
+// every time a new IP is discovered. The Probes returned will be closed if an
 // IP address is no longer in the DNS record for the given hostname. This can
 // be used to healthcheck every IP address that a hostname resolves to.
-func ForEachAddr(host string, newProbe func(netip.Addr) *Probe, opts ForEachAddrOpts) ProbeFunc {
-	return makeForEachAddr(host, newProbe, opts).run
+func ForEachAddr(host string, makeProbes func(netip.Addr) []*Probe, opts ForEachAddrOpts) ProbeFunc {
```

Comment:
if caller has to pass this struct value anyway, maybe move the func to an opt?

Reply:
All of the "options" are actually optional (i.e. you can pass a zero-valued `opts` struct), whereas both `host` and `makeProbes` are required so I made them actual parameters.